src/Compilers/VisualBasic/Test/Semantic/Diagnostics/GetDiagnosticsTests.vb

```diff
@@ -224,6 +224,145 @@ End Namespace
+            Dim parseOptions2 = New VisualBasicParseOptions().WithPreprocessorSymbols(dict2)
+
+            Dim syntaxTree1 = Parse(String.Empty, options:=parseOptions1)
+            Dim syntaxTree2 = Parse(String.Empty, options:=parseOptions2)
+            Dim syntaxTree3 = Parse(String.Empty, options:=parseOptions2)
+
+            Dim options = New VisualBasicCompilationOptions(OutputKind.DynamicallyLinkedLibrary, concurrentBuild:=concurrentBuild)
+            Dim Compilation = CreateCompilationWithMscorlib({syntaxTree1, syntaxTree2, syntaxTree3}, options:=options)
+
+            CompilationUtils.AssertTheseDiagnostics(Compilation,
+<errors>
+BC31030: Conditional compilation constant '1' is not valid: Identifier expected.
+BC31030: Conditional compilation constant '2' is not valid: Identifier expected.
+</errors>)
+        End Sub
+
+        <Theory>
+        <InlineData(True)>
+        <InlineData(False)>
+        Public Sub DiagnositcsInCompilationOptionsParseOptionsAreDedupedWithParseTreesParseOptions(concurrentBuild As Boolean)
+            Dim dict1 = New Dictionary(Of String, Object)
+            dict1.Add("1", Nothing)
+            Dim dict2 = New Dictionary(Of String, Object)
+            dict2.Add("2", Nothing)
+
+            Dim parseOptions1 = New VisualBasicParseOptions().WithPreprocessorSymbols(dict1)
+            Dim parseOptions2 = New VisualBasicParseOptions().WithPreprocessorSymbols(dict2)
+
+            Dim syntaxTree1 = Parse(String.Empty, options:=parseOptions1)
+            Dim syntaxTree2 = Parse(String.Empty, options:=parseOptions2)
+
+            Dim options = New VisualBasicCompilationOptions(OutputKind.DynamicallyLinkedLibrary, concurrentBuild:=concurrentBuild, parseOptions:=parseOptions1)
+            Dim Compilation = CreateCompilationWithMscorlibAndVBRuntime({syntaxTree1, syntaxTree2}, options:=options)
+
+            CompilationUtils.AssertTheseDiagnostics(Compilation,
+<errors>
+BC31030: Conditional compilation constant '1' is not valid: Identifier expected.
+BC31030: Conditional compilation constant '2' is not valid: Identifier expected.
+</errors>)
+        End Sub
+
+        <Theory>
+        <InlineData(True)>
+        <InlineData(False)>
+        Public Sub DiagnositcsInCompilationOptionsParseOptionsAreReportedSeparately(concurrentBuild As Boolean)
+            Dim dict1 = New Dictionary(Of String, Object)
+            dict1.Add("1", Nothing)
+            Dim dict2 = New Dictionary(Of String, Object)
+            dict2.Add("2", Nothing)
+
+            Dim parseOptions1 = New VisualBasicParseOptions().WithPreprocessorSymbols(dict1)
+            Dim parseOptions2 = New VisualBasicParseOptions().WithPreprocessorSymbols(dict2)
+
+            Dim syntaxTree1 = Parse(String.Empty, options:=parseOptions1)
+
+            Dim options = New VisualBasicCompilationOptions(OutputKind.DynamicallyLinkedLibrary, concurrentBuild:=concurrentBuild, parseOptions:=parseOptions2)
```

Comment:
>Dim options = New VisualBasicCompilationOptions(OutputKind.DynamicallyLinkedLibrary, concurrentBuild:=concurrentBuild, parseOptions:=parseOptions2) [](start = 12, length = 147)

We should assert errors from compilation options, unless we have another unit-test that is doing this.